### ISSUE TITLE
statsmodels.graphics.functional pickle issue

### ISSUE THREAD

**Issue by EricWrightAtWork** [statsmodels/statsmodels#4772]
I have code that needs to pickle functions (i.e. for joblib) that are defined in modules where statsmodels.api is imported. The pickle attempt leads to an `AttributeError` from the `statsmodels.graphics.functional` module:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\[...]\lib\site-packages\statsmodels\graphics\functional.py", line 32, in _pickle_method
    if m.im_self is None:
AttributeError: 'function' object has no attribute 'im_self'
```
This can be simply reproduced by creating a short module with the following code:
```
import statsmodels.api as sm
class Dummy(object):
    def __init__(self):
        pass
    def get_add_constant(self):
        return sm.add_constant
```
If this code is saved in a file, `test_mod.py`, then I can trigger the exception by opening a python session and entering:
```
>>> from test_mod import Dummy
>>> dum = Dummy()
>>> pickle.dumps(dum.get_add_constant)
```

I have worked around the issue by narrowing my import to just the classes needed in my code, but this seems like a bug in graphics.functional that should be addressed. 

Note: I observed this issue using statsmodels 0.9.0 in Windows.

**Comment by josef-pkt** [comment]
@EricWrightAtWork  Thanks for reporting.

I can replicate it as soon as statsmodels.api is imported, even when it is not used.
That is, copyreg in graphics changes the global registry which messes up pickling in other parts of statsmodels.

I don't know yet what the fix is, 
Do we adjust the pickle_method in graphics functional or do we drop it completely?
I don't think some specific module should change globals or global settings, i.e. if we need to change a global registry, then it should go into tools like exceptions and warnings.

**Comment by pseudotensor** [comment]
I get same error for 0.9.0 but not 0.8.0.

**Comment by jfindlay** [comment]
For anyone who encounters this bug, here is a workaround I am using for 0.9.0:
```py
# Workaround https://github.com/statsmodels/statsmodels/issues/4772.  Remove
# when no longer using statsmodels<=0.9.0.
def fix_statsmodels_pickle():
    try:
        import copyreg
    except ImportError:
        import copy_reg as copyreg
    import types

    if hasattr(copyreg, 'dispatch_table') \
    and isinstance(copyreg.dispatch_table, dict) \
    and types.MethodType in copyreg.dispatch_table:
        method_pickler = copyreg.dispatch_table.get(types.MethodType)
        if hasattr(method_pickler, '__module__') \
        and 'statsmodels.graphics.functional' in method_pickler.__module__ \
        and hasattr(method_pickler, '__name__') \
        and '_pickle_method' in method_pickler.__name__:
            del copyreg.dispatch_table[types.MethodType]
```
```py
import statsmodels.graphics.functional
import ftk.compat ; ftk.compat.fix_statsmodels_pickle()
```

**Comment by rebecca-palmer** [comment]
pip has 0.10 which includes this fix; no incompatible changes are specifically listed in the release notes, but I haven't checked further.

The fix (#4963) looks like it may well work in 0.9, but I haven't had time to test it.
